articles/quickstart/backend/java-spring-security5/01-authorization.md

```diff
@@ -0,0 +1,309 @@
+    private final String audience;
+
+    AudienceValidator(String audience) {
+        this.audience = audience;
+    }
+
+    public OAuth2TokenValidatorResult validate(Jwt jwt) {
+        OAuth2Error error = new OAuth2Error("invalid_token", "The required audience is missing", null);
+        
+        if (jwt.getAudience().contains(audience)) {
+            return OAuth2TokenValidatorResult.success();
+        }
+        return OAuth2TokenValidatorResult.failure(error);
+    }
+}
+```
+
+Update the `SecurityConfig` class to configure a `JwtDecoder` bean that uses the `AudienceValidator`:
+
+```java
+// src/main/java/com/auth0/example/security/SecurityConfig.java
+
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Value("<%= "${auth0.audience}" %>")
+    private String audience;
+
+    @Value("<%= "${spring.security.oauth2.resourceserver.jwt.issuer-uri}" %>")
+    private String issuer;
+
+    @Bean
+    JwtDecoder jwtDecoder() {
+        NimbusJwtDecoderJwkSupport jwtDecoder = (NimbusJwtDecoderJwkSupport)
+                JwtDecoders.fromOidcIssuerLocation(issuer);
+
+        OAuth2TokenValidator<Jwt> audienceValidator = new AudienceValidator(audience);
+        OAuth2TokenValidator<Jwt> withIssuer = JwtValidators.createDefaultWithIssuer(issuer);
+        OAuth2TokenValidator<Jwt> withAudience = new DelegatingOAuth2TokenValidator<>(withIssuer, audienceValidator);
+
+        jwtDecoder.setJwtValidator(withAudience);
+
+        return jwtDecoder;
+    }
+}
+```
+
+## Protect API Endpoints
+
+The routes shown below are available for the following requests:
+
+* `GET /api/public`: available for non-authenticated requests
+* `GET /api/private`: available for authenticated requests containing a valid JWT Access Token
+* `GET /api/private-scoped`: available for authenticated requests containing a valid JWT Access Token with the `read:messages` scope granted
+
+The example below shows how to secure API methods using the `HttpSecurity` provided in the `configure()` method of the `SecurityConfig` class. Route matchers are used to restrict access based on the level of authorization requred:
+
+```java
+// src/main/java/com/auth0/example/security/SecurityConfig.java
+
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Override
+    public void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .mvcMatchers("/api/public").permitAll()
+                .mvcMatchers("/api/private").authenticated()
+                .mvcMatchers("api/private-scoped").hasAuthority("SCOPE_read:messages")
+                .and()
+                .oauth2ResourceServer().jwt();
+    }
+}
+```
+
+::: note
+By default, Spring Security will create a `GrantedAuthority` for each scope in the `scope` claim of the JWT. This is what enables using the `hasAuthority("SCOPE_read:messages")` method to restrict access to a valid JWT that contains the `read:messages` scope.
```

Comment:
`GrantedAuthority` is an interface, so it could be "Spring Security will create an instance of `GrantedAuthority`...". However, that reads a little long and is kind of redundant IMO. For those familiar with Spring and Spring Security, I think it would read normally.